src/providers/LedgerProvider.js

```diff
@@ -44,7 +44,7 @@ export default class LedgerProvider extends Provider {
 
     while (!path) {
       const addr = await this.getAddresses(index, 1)
-      if (String(addr) === address) path = this.getDerivationPathFromIndex(index)
+      if (String(addr[0].address) === address) path = this.getDerivationPathFromIndex(index)
```

Comment:
You can just use `String(addr[0])`. All `Address` objects have `.toString()` methods.

Reply:
Good catch. Reverted